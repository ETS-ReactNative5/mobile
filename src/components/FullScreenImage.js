import React from 'react'
import {
  Modal,
  TouchableOpacity,
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import ZoomableImage from './ZoomableImage'
import Icon from 'react-native-vector-icons/FontAwesome'
import PropTypes from 'prop-types';

class FullScreenImage extends React.Component {
  render() {
    return (
      <Modal
        animationType={'fade'}
        transparent={true}
        onRequestClose={() => {}}
        visible={this.props.isVisible}>
        <View style={styles.container}>
          <ZoomableImage
            source={this.props.source}
            handlePress={this.props.handlePress}
          />
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={this.props.handlePress}
            style={styles.closeIcon}>
            <Icon name="times" style={styles.icon} />
          </TouchableOpacity>
        </View>
      </Modal>
    );
  }
}

const styles = EStyleSheet.create({
  container: {
    backgroundColor: 'rgba(0, 0, 0, 0.9)',
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  closeIcon: {
    backgroundColor: 'transparent',
    position: 'absolute',
    top: 13,
    right: 5
  },
  icon: {
    backgroundColor: 'transparent',
    color: 'white',
    fontSize: 24,
    padding: 15,
  },
  rowContainer: {
    backgroundColor: 'rgba(0, 0, 0, 0.9)',
    position: 'absolute',
    bottom: 10,
    left: 10,
    flexDirection: 'row',
    alignItems: 'center'
  },
  message: {
    color: 'white',
  },
  infoIcon: {
    color: '$transluscentWhite',
    fontSize: 20,
    padding: 5,
  }
})

FullScreenImage.propTypes = {
  source: PropTypes.object,
  isVisible: PropTypes.bool,
  handlePress: PropTypes.func,
}

export default FullScreenImage
