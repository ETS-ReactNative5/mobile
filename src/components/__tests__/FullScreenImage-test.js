import 'react-native'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import FullScreenImage from '../FullScreenImage'

const imageSource = {
  uri: 'https://placekitten.com/200/300'
}

jest.mock('../ZoomableImage', () => 'ZoomableImage')

const renderer = new ShallowRenderer();

it('renders correctly', () => {
  const tree = renderer.render(
    <FullScreenImage
      source={imageSource}
      handlePress={jest.fn}
      isVisible={true}
      allowPanAndZoom={true}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('renders hidden correctly', () => {
  const tree = renderer.render(
    <FullScreenImage
      source={imageSource}
      handlePress={jest.fn}
      isVisible={false}
      allowPanAndZoom={true}
    />
  )
  expect(tree).toMatchSnapshot()
})
