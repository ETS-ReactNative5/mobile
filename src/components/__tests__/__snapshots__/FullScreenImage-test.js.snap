// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View>
    <Connect(ZoomableImage)
      handlePress={[Function]}
      source={
        Object {
          "uri": "https://placekitten.com/200/300",
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="times"
        size={12}
      />
    </TouchableOpacity>
  </View>
</Component>
`;

exports[`renders hidden correctly 1`] = `
<Component
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View>
    <Connect(ZoomableImage)
      handlePress={[Function]}
      source={
        Object {
          "uri": "https://placekitten.com/200/300",
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="times"
        size={12}
      />
    </TouchableOpacity>
  </View>
</Component>
`;
